### PR TITLE
nbhood clizification tidy-up

### PR DESCRIPTION
**Pull request by bayliffe**
Changes following an inadvertently skipped review.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/nbhood/__init__.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/nbhood.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4777242&bid=15917255)

File: lib/improver/cli/nbhood.py [hunk 1]
```diff
@@ -149,9 +143,6 @@ def process(cube: cli.inputcube,
     from improver.wind_calculations.wind_direction import WindDirection
 
     if neighbourhood_output == "percentiles":
-        if neighbourhood_shape == "square":
-            raise RuntimeError('neighbourhood_shape="square" cannot be used '
-                               'with neighbourhood_output="percentiles"')
```
Comment (MoseleyS) [inline lib/improver/cli/nbhood.py]: The corresponding `Raises` entries also need deleting from the doc-string.

Comment (bayliffe) [reply lib/improver/cli/nbhood.py]: Done.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/nbhood/__init__.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/nbhood.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4777242&bid=15917255)

**Pull request merged**